crates/cubecl-core/src/frontend/synchronization.rs

```diff
@@ -13,16 +14,47 @@ use crate::{
 /// * Then all the invocations in the cube wait for each other to arrive at the barrier, i.e. this step.
 ///
 /// * Then all the invocations int the cube begin executing after the barrier, and all writes to cube address space made before the barrier are now visible to any invocation in this cube.
-pub fn sync_units() {}
+pub fn sync_cube() {}
 
-pub mod sync_units {
+pub mod sync_cube {
     use super::*;
 
     pub fn expand(scope: &mut Scope) {
-        scope.register(Synchronization::SyncUnits)
+        scope.register(Synchronization::SyncCube)
     }
 }
 
+/// Synchronizes threads within their plane (e.g., warp or SIMD group).
+///
+/// Not all targets support plane-level synchronization. Use [`SyncPlaneMode`]
+/// to control whether a fallback to cube-wide synchronization is allowed
+/// when plane sync is unavailable.
+pub fn sync_plane(_sync_plane_mode: SyncPlaneMode) {
+    unexpanded!()
+}
+
+pub mod sync_plane {
+    use super::*;
+
+    pub fn expand(scope: &mut Scope, sync_plane_mode: SyncPlaneMode) {
+        match sync_plane_mode {
+            SyncPlaneMode::Strict => scope.register(Synchronization::SyncPlaneStrict),
+            SyncPlaneMode::Fallback => scope.register(Synchronization::SyncPlaneFallback),
+        }
+    }
+}
+
+/// Determines whether a sync plane is allowed to fall back to a sync cube
+/// when native support is unavailable.
+///
+/// In some cases, falling back may only incur a minor performance cost.
+/// In others, it can break correctness or even cause deadlocks —
+/// especially when warp specialization is involved.
+pub enum SyncPlaneMode {
+    Strict,   // Requires native warp sync, or fails at kernel compilation
+    Fallback, // Allows fallback to Cube-wide sync, may break correctness
+}
+
```

Comment:
Not sure I would allow a fallback in the `sync_plane` function. If you need some in your context. You may create a wrapper for your use case with a comptime value passed as input to determine which sync function to use.